shell/platform/darwin/ios/framework/Source/SemanticsObject+UIFocusSystem.mm

```diff
@@ -48,6 +55,9 @@ - (void)didUpdateFocusInContext:(UIFocusUpdateContext*)context
 }
 
 - (id<UIFocusEnvironment>)parentFocusEnvironment {
+  if ([self.parent isKindOfClass:[FlutterScrollableSemanticsObject class]]) {
```

Comment:
doesn't this need to be recursively up?

Reply:
since we recursively looks up in frame calculation

Reply:
What do you mean by recursively up? This method is supposed to return the parent object in the focus hierarchy not the root (this method is part of the `UIFocusEnvironment` protocol and the iOS focus engine will use this method to traverse the focus hierarchy).